### PR TITLE
Mon 3112 fix source install

### PR DESCRIPTION
**Pull request by sc979**
Resolve : MON-3112

### CODE REVIEW

File: doc/en/installation/from_sources.rst [hunk 1]
```diff
@@ -67,25 +67,83 @@ Then execute::
...
+
+    $ apt-get install apt-transport-https lsb-release ca-certificates
+    $ wget -O /etc/apt/trusted.gpg.d/php.gpg https://packages.sury.org/php/apt.gpg
+    $ echo "deb https://packages.sury.org/php/ $(lsb_release -sc) main" >> /etc/apt/sources.list.d/php.list
+    $ apt-get update
+
+For Ubuntu 18.04 :
 .. note::
-   Debian and Ubuntu latest version not yet supported.
+It is necessary to add sudo in front of the following commands
```
Comment (chgautier) [inline doc/en/installation/from_sources.rst]:
```suggestion
You must add sudo in front of the following commands
```

File: doc/en/installation/from_sources.rst [hunk 2]
```diff
@@ -154,19 +212,30 @@ Shell Installation
...
+    $ ./install.sh -i
 
 .. note::
 
- The installation script allows customised configuration; this process will show you the best paths to use. Furthermore quick yes/no questions can be replied to by [y] most of the time.
+  The installation script allows customized configuration; this process will show you the best paths to use. Furthermore quick yes/no questions can be replied to by [y] most of the time.
+
+.. note::
+
+  If centreon sources have been downloaded from github, run those commands :
```
Comment (chgautier) [inline doc/en/installation/from_sources.rst]:
```suggestion
  If centreon sources have been downloaded from github, run the following commands :
```

File: libinstall/CentWeb.sh [hunk 3]
```diff
@@ -18,7 +18,11 @@ echo -e "$line"
 
 ###### check space ton tmp dir
 check_tmp_disk_space
-[ "$?" -eq 1 ] && purge_centreon_tmp_dir
+if [ "$silent_install" -eq 1 ] ; then
+  [ "$?" -eq 1 ] && purge_centreon_tmp_dir "silent"
```
Comment (ganoze) [inline libinstall/CentWeb.sh]: This should likely be modified to the following.

```
check_tmp_disk_space
if [ "$?" -eq 1 ] ; then
  if [ "$silent_install" -eq 1 ] ; then
    purge_centreon_tmp_dir "silent"
  else
    purge_centreon_tmp_dir
  fi
fi
```

File: libinstall/CentWeb.sh [hunk 4]
```diff
@@ -120,24 +129,36 @@ $INSTALL_DIR/cinstall $cinstall_opts \
     "$CENTREON_ETC" >> "$LOG_FILE" 2>&1
 check_result $? "$(gettext "Change right on") $CENTREON_ETC"
 
+## Generate translation
```
Comment (ganoze) [inline libinstall/CentWeb.sh]: This is supposed to be handled at source creation.

File: libinstall/functions [hunk 5]
```diff
@@ -216,9 +216,9 @@ function yes_no_default() {
 		read res
 		[ -z "$res" ] && res="$default"
 	done
-	if [ "$res" = "$yes" ] ; then
+	if [ "$res" = "$yes" ] ; then
```
Comment (ganoze) [inline libinstall/functions]: Please remove superfluous whitespaces.

File: libinstall/functions [hunk 6]
```diff
@@ -232,12 +232,12 @@ function answer() {
 	local var_ref=$3
 	local res=""
 	local first=0
-	while [ -z "$res" ] ; do
+	while [ ! -z "$res" ] ; do
```
Comment (ganoze) [inline libinstall/functions]: ! is a shell metachar. Could you escape it ?

File: libinstall/functions [hunk 7]
```diff
@@ -232,12 +232,12 @@ function answer() {
...
 		echo -e "\n$message"
 		[ "$default" != "NO_DEFAULT" ] && echo -e "$(gettext "default to [$default]")"
 		echo -en "> "
 		read res
-		if [ -z "$res" ] ; then
+		if [ -z "$res" ] ; then
```
Comment (ganoze) [inline libinstall/functions]: Superfluous end-of-line whitespace.

File: libinstall/functions [hunk 8]
```diff
@@ -307,7 +307,7 @@ function answer_with_createdir() {
 			res="not_define"
 		else
 			first=1
-			[ -d "$res" ] && break
+			[ -d "$res" ] && break
```
Comment (ganoze) [inline libinstall/functions]: Superfluous end-of-line whitespace.

File: libinstall/functions [hunk 9]
```diff
@@ -529,7 +529,7 @@ function testfile_clean() {
 	if [ ! -e "$file" ] ; then
 		eval $var=""
 		return 1
-	else
+	else
```
Comment (ganoze) [inline libinstall/functions]: Superfluous end-of-line whitespace.

File: libinstall/functions [hunk 10]
```diff
@@ -994,11 +968,11 @@ function locate_sudo() {
 #----
 function locate_rrdtool() {
 	testfile_clean "$BIN_RRDTOOL" "BIN_RRDTOOL"
-	if [ -z "$BIN_RRDTOOL" ] ; then
+	if [ -z "$BIN_RRDTOOL" ] ; then
```
Comment (ganoze) [inline libinstall/functions]: Superfluous end-of-line whitespace.

File: libinstall/functions [hunk 11]
```diff
@@ -994,11 +968,11 @@ function locate_sudo() {
 #----
 function locate_rrdtool() {
 	testfile_clean "$BIN_RRDTOOL" "BIN_RRDTOOL"
-	if [ -z "$BIN_RRDTOOL" ] ; then
+	if [ -z "$BIN_RRDTOOL" ] ; then 
 		pathfind_ret "rrdtool" "BIN_RRDTOOL"
-		if [ "$?" -ne 0 ] ; then
+		if [ "$?" -ne 0 ] ; then
```
Comment (ganoze) [inline libinstall/functions]: Superfluous end-of-line whitespace.

File: libinstall/functions [hunk 12]
```diff
@@ -994,11 +968,11 @@ function locate_sudo() {
...
 function locate_rrdtool() {
 	testfile_clean "$BIN_RRDTOOL" "BIN_RRDTOOL"
-	if [ -z "$BIN_RRDTOOL" ] ; then
+	if [ -z "$BIN_RRDTOOL" ] ; then 
 		pathfind_ret "rrdtool" "BIN_RRDTOOL"
-		if [ "$?" -ne 0 ] ; then
+		if [ "$?" -ne 0 ] ; then 
 			answer_with_testfile "$(gettext "Where is rrdtool")" "$DEFAULT_BIN_RRDTOOL" "BIN_RRDTOOL"
-		else
+		else
```
Comment (ganoze) [inline libinstall/functions]: Superfluous end-of-line whitespace.

File: libinstall/functions [hunk 13]
```diff
@@ -1015,11 +989,11 @@ function locate_rrdtool() {
 #----
 function locate_perl() {
 	testfile_clean "$BIN_PERL" "BIN_PERL"
-	if [ -z "$BIN_PERL" ] ; then
+	if [ -z "$BIN_PERL" ] ; then
```
Comment (ganoze) [inline libinstall/functions]: Superfluous end-of-line whitespace.

File: libinstall/functions [hunk 14]
```diff
@@ -1015,11 +989,11 @@ function locate_rrdtool() {
 #----
 function locate_perl() {
 	testfile_clean "$BIN_PERL" "BIN_PERL"
-	if [ -z "$BIN_PERL" ] ; then
+	if [ -z "$BIN_PERL" ] ; then 
 		pathfind_ret "perl" "BIN_PERL"
-		if [ "$?" -ne 0 ] ; then
+		if [ "$?" -ne 0 ] ; then
```
Comment (ganoze) [inline libinstall/functions]: Superfluous end-of-line whitespace.

File: libinstall/functions [hunk 15]
```diff
@@ -1015,11 +989,11 @@ function locate_rrdtool() {
...
 function locate_perl() {
 	testfile_clean "$BIN_PERL" "BIN_PERL"
-	if [ -z "$BIN_PERL" ] ; then
+	if [ -z "$BIN_PERL" ] ; then 
 		pathfind_ret "perl" "BIN_PERL"
-		if [ "$?" -ne 0 ] ; then
+		if [ "$?" -ne 0 ] ; then 
 			answer_with_testfile "$(gettext "Where is perl")" "$DEFAULT_BIN_PERL" "BIN_PERL"
-		else
+		else
```
Comment (ganoze) [inline libinstall/functions]: Superfluous end-of-line whitespace.

File: libinstall/functions [hunk 16]
```diff
@@ -1036,11 +1010,11 @@ function locate_perl() {
 #----
 function locate_mail() {
 	testfile_clean "$BIN_MAIL" "BIN_MAIL"
-	if [ -z "$BIN_MAIL" ] ; then
+	if [ -z "$BIN_MAIL" ] ; then
```
Comment (ganoze) [inline libinstall/functions]: Superfluous end-of-line whitespace.

File: libinstall/functions [hunk 17]
```diff
@@ -1036,11 +1010,11 @@ function locate_perl() {
 #----
 function locate_mail() {
 	testfile_clean "$BIN_MAIL" "BIN_MAIL"
-	if [ -z "$BIN_MAIL" ] ; then
+	if [ -z "$BIN_MAIL" ] ; then 
 		pathfind_ret "mail" "BIN_MAIL"
-		if [ "$?" -ne 0 ] ; then
+		if [ "$?" -ne 0 ] ; then
```
Comment (ganoze) [inline libinstall/functions]: Superfluous end-of-line whitespace.

File: libinstall/functions [hunk 18]
```diff
@@ -1036,11 +1010,11 @@ function locate_perl() {
...
 function locate_mail() {
 	testfile_clean "$BIN_MAIL" "BIN_MAIL"
-	if [ -z "$BIN_MAIL" ] ; then
+	if [ -z "$BIN_MAIL" ] ; then 
 		pathfind_ret "mail" "BIN_MAIL"
-		if [ "$?" -ne 0 ] ; then
+		if [ "$?" -ne 0 ] ; then 
 			answer_with_testfile "$(gettext "Where is mail binary")" "$DEFAULT_BIN_MAIL" "BIN_MAIL"
-		else
+		else
```
Comment (ganoze) [inline libinstall/functions]: Superfluous end-of-line whitespace.

File: libinstall/functions [hunk 19]
```diff
@@ -1193,34 +1180,34 @@ function configureApache() {
 	# Generate a apache config in directory
 	prepare_apache_config "$dir_conf"
 
-	if [ "$silent_install" -ne 1 ] ; then
+	if [ "$silent_install" -ne 1 ] ; then
```
Comment (ganoze) [inline libinstall/functions]: Superfluous end-of-line whitespace.

File: libinstall/functions [hunk 20]
```diff
@@ -1193,34 +1180,34 @@ function configureApache() {
...
 	prepare_apache_config "$dir_conf"
 
-	if [ "$silent_install" -ne 1 ] ; then
+	if [ "$silent_install" -ne 1 ] ; then 
 		if [ -e $DIR_APACHE_CONF/centreon.conf ] ; then
 			echo "$(gettext "Finding Apache Centreon configuration file")"
 			echo_success "'$DIR_APACHE_CONF/centreon.conf' :" "$ok"
 			yes_no_default "$(gettext "Do you want to update Centreon Apache sub configuration file ?")"
-                        if [ "$?" -eq 0 ] ; then
+                        if [ "$?" -eq 0 ] ; then
```
Comment (ganoze) [inline libinstall/functions]: Superfluous end-of-line whitespace.

File: libinstall/functions [hunk 21]
```diff
@@ -1193,34 +1180,34 @@ function configureApache() {
...
 			echo_success "'$DIR_APACHE_CONF/centreon.conf' :" "$ok"
 			yes_no_default "$(gettext "Do you want to update Centreon Apache sub configuration file ?")"
-                        if [ "$?" -eq 0 ] ; then
+                        if [ "$?" -eq 0 ] ; then 
                             write_config="1"
                             cp -f $DIR_APACHE_CONF/centreon.conf $DIR_APACHE_CONF/centreon.conf-bak >> $LOG_FILE 2>&1
                             echo_info "$(gettext "Backup Centreon Apache configuration completed")"
                         fi
-		else
+		else
```
Comment (ganoze) [inline libinstall/functions]: Superfluous end-of-line whitespace.

File: libinstall/functions [hunk 22]
```diff
@@ -1193,34 +1180,34 @@ function configureApache() {
...
                             cp -f $DIR_APACHE_CONF/centreon.conf $DIR_APACHE_CONF/centreon.conf-bak >> $LOG_FILE 2>&1
                             echo_info "$(gettext "Backup Centreon Apache configuration completed")"
                         fi
-		else
+		else 
 			yes_no_default "$(gettext "Do you want to add Centreon Apache sub configuration file ?")"
 			[ "$?" -eq 0 ] && write_config="1"
 		fi
-	else
+	else
```
Comment (ganoze) [inline libinstall/functions]: Superfluous end-of-line whitespace.

File: libinstall/functions [hunk 23]
```diff
@@ -1193,34 +1180,34 @@ function configureApache() {
...
+		else 
 			yes_no_default "$(gettext "Do you want to add Centreon Apache sub configuration file ?")"
 			[ "$?" -eq 0 ] && write_config="1"
 		fi
-	else
+	else 
 		write_config="1"
     	fi
-	if [ "$write_config" -eq 1 ] ; then
+	if [ "$write_config" -eq 1 ] ; then
```
Comment (ganoze) [inline libinstall/functions]: Superfluous end-of-line whitespace.

File: libinstall/functions [hunk 24]
```diff
@@ -1193,34 +1180,34 @@ function configureApache() {
...
+	if [ "$write_config" -eq 1 ] ; then 
                 if [ "$upgrade" -eq 1 ] ; then
                     cp -f $dir_conf/centreon.apache.conf $DIR_APACHE_CONF/centreon.conf.new >> $LOG_FILE 2>&1
                 else
                     cp -f $dir_conf/centreon.apache.conf $DIR_APACHE_CONF/centreon.conf >> $LOG_FILE 2>&1
                 fi
 		echo_success "$(gettext "Create") '$DIR_APACHE_CONF/centreon.conf'" "$ok"
 		echo_success "$(gettext "Configuring Apache")" "$ok"
-		# After finishing the configuration ->
+		# After finishing the configuration ->
```
Comment (ganoze) [inline libinstall/functions]: Superfluous end-of-line whitespace.

File: libinstall/functions [hunk 25]
```diff
@@ -1193,34 +1180,34 @@ function configureApache() {
...
                 else
                     cp -f $dir_conf/centreon.apache.conf $DIR_APACHE_CONF/centreon.conf >> $LOG_FILE 2>&1
                 fi
 		echo_success "$(gettext "Create") '$DIR_APACHE_CONF/centreon.conf'" "$ok"
 		echo_success "$(gettext "Configuring Apache")" "$ok"
-		# After finishing the configuration ->
+		# After finishing the configuration -> 
 		# reload apache ! (if admin wants :p)
-		if [ "${APACHE_RELOAD:-0}" -eq 0 ] ; then
+		if [ "${APACHE_RELOAD:-0}" -eq 0 ] ; then
```
Comment (ganoze) [inline libinstall/functions]: Superfluous end-of-line whitespace.

File: libinstall/functions [hunk 26]
```diff
@@ -1433,19 +1533,19 @@ function is_suspect_conf_sudo() {
 ## @Globals	SUDO_FILE, line
 ## @return 0	end
 #----
-function configureSUDO(){
+function configureSUDO() {
 	echo -e "\n$line"
 	echo -e "\t$(gettext "Configure Sudo")"
 	echo -e "$line"
-    	local no_force_sudo="0"
+    local no_force_sudo="0"
```
Comment (ganoze) [inline libinstall/functions]: Inconsistent padding.

File: libinstall/functions [hunk 27]
```diff
@@ -1433,19 +1533,19 @@ function is_suspect_conf_sudo() {
...
 	local dir_conf="$1"
 	# Generate a sudo config in directory
 	prepare_sudo_config "$dir_conf"
 
 	# Try to detect old configuration
 	sudo=`${GREP} -e "CENTREON SUDO" $SUDO_FILE > /dev/null 2>&1 ; echo $?`
 	if [ $sudo -eq 1 ]; then
-    		echo_info "$(gettext "Your sudo is not configured")"
+    	echo_info "$(gettext "Your sudo is not configured")"
```
Comment (ganoze) [inline libinstall/functions]: Inconsistent padding.

File: libinstall/CentWeb.sh [hunk 28]
```diff
@@ -126,18 +137,26 @@ cp -Rf $TMP_DIR/src/www $TMP_DIR/final
 cp -Rf $TMP_DIR/src/GPL_LIB $TMP_DIR/final
 mkdir -p $TMP_DIR/final/config
 cp -Rf $TMP_DIR/src/config/partition.d $TMP_DIR/final/config/partition.d
-cp -f $TMP_DIR/src/config/centreon.config.php.template $TMP_DIR/final/config/centreon.config.php
+mv $TMP_DIR/src/config/centreon.config.php.template $TMP_DIR/src/config/centreon.config.php
```
Comment (ganoze) [inline libinstall/CentWeb.sh]: Isn't it supposed to be cp'd to the _/final_ subdirectory ?

File: libinstall/CentPluginsTraps.sh [hunk 29]
```diff
@@ -15,7 +15,11 @@ echo -e "$line"
 
 ###### Check disk space
 check_tmp_disk_space
-[ "$?" -eq 1 ] && purge_centreon_tmp_dir
+if [ "$silent_install" -eq 1 ] ; then
```
Comment (ganoze) [inline libinstall/CentPluginsTraps.sh]: Please use the same construct than in _libinstall/CentWeb.sh_.

File: libinstall/CentWeb.sh [hunk 30]
```diff
@@ -327,15 +346,19 @@ $INSTALL_DIR/cinstall $cinstall \
     $INSTALL_DIR_CENTREON/www >> "$LOG_FILE" 2>&1
 
 $INSTALL_DIR/cinstall $cinstall_opts \
-    -u "$CENTREON_USER" -g "$CENTREON_GROUP" -d 755 -m 644 \
+    -u "$CENTREON_USER" -g "$CENTREON_GROUP" -d 775 -m 775 \
```
Comment (ganoze) [inline libinstall/CentWeb.sh]: Wow ! Permission changes on files seem excessively permissives !

File: libinstall/CentWeb.sh [hunk 31]
```diff
@@ -327,15 +346,19 @@ $INSTALL_DIR/cinstall $cinstall \
...
     -p $TMP_DIR/final/www \
     $TMP_DIR/final/www/* $INSTALL_DIR_CENTREON/www/ >> "$LOG_FILE" 2>&1
 check_result $? "$(gettext "Install CentWeb (web front of centreon)")"
 
+cp -f $TMP_DIR/final/www/.htaccess $INSTALL_DIR_CENTREON/www/ >> "$LOG_FILE" 2>&1
+$CHOWN $WEB_USER:$WEB_GROUP $INSTALL_DIR_CENTREON/www/.htaccess
+$CHMOD 775 $INSTALL_DIR_CENTREON/www/.htaccess
```
Comment (ganoze) [inline libinstall/CentWeb.sh]: Wouldn't 644 be sufficient ?

File: libinstall/CentWeb.sh (hunk 30 above)
Comment (sc979) [reply libinstall/CentWeb.sh]: it's needed for the debian install

File: libinstall/CentWeb.sh (hunk 31 above)
Comment (sc979) [reply libinstall/CentWeb.sh]: nope, I don't think so

File: libinstall/CentWeb.sh (hunk 30 above)
Comment (ganoze) [reply libinstall/CentWeb.sh]: Could you please explain what is the specific issue with the existing permissions ?

File: libinstall/CentWeb.sh (hunk 31 above)
Comment (ganoze) [reply libinstall/CentWeb.sh]: Could you explain why a configuration file needs +x ?

And why does the group need +w ? Is .htaccess overwritten by any application ?

File: libinstall/CentWeb.sh (hunk 30 above)
Comment (sc979) [reply libinstall/CentWeb.sh]: the .htaccess is modified somewhere else later... I can't find where.
But if I set it to 644, I'm stuck on the login screen and I can't access the IHM.

File: libinstall/CentWeb.sh (hunk 31 above)
Comment (sc979) [reply libinstall/CentWeb.sh]: the .htaccess is modified somewhere else later... I can't find where.
But if I set it to 644, I'm stuck on the login screen and I can't access the IHM.

File: libinstall/CentPlugins.sh [hunk 32]
```diff
@@ -15,7 +15,13 @@ echo -e "$line"
 
 ###### Check disk space
 check_tmp_disk_space
-[ "$?" -eq 1 ] && purge_centreon_tmp_dir
+if [ "$?" -eq 1 ] ; then
```
Comment (ganoze) [inline libinstall/CentPlugins.sh]: Please remove the entire CentPlugins.sh. It seems like it is not needed anymore.

File: libinstall/CentWeb.sh [hunk 33]
```diff
@@ -327,15 +346,19 @@ $INSTALL_DIR/cinstall $cinstall \
     $INSTALL_DIR_CENTREON/www >> "$LOG_FILE" 2>&1
 
 $INSTALL_DIR/cinstall $cinstall_opts \
-    -u "$CENTREON_USER" -g "$CENTREON_GROUP" -d 755 -m 644 \
+    -u "$CENTREON_USER" -g "$CENTREON_GROUP" -d 775 -m 644 \
```
Comment (ganoze) [inline libinstall/CentWeb.sh]: Permissive permission.

File: libinstall/CentPlugins.sh (hunk 32 above)
Comment (sc979) [reply libinstall/CentPlugins.sh]: please check "install.sh +406"

**Pull request merged**